editor/store/reducer.js

```diff
@@ -485,6 +485,64 @@ export function blocksMode( state = {}, action ) {
 	return state;
 }
 
+/**
+ * Reducer returning the currently autosaving status.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Dispatched action
+ * @return {Object}        Updated state
+ */
+export function currentlyAutosaving( state = false, action ) {
+	switch ( action.type ) {
+		case 'DOING_AUTOSAVE':
+			const { isAutosaving } = action;
+			return isAutosaving;
+	}
+
+	return state;
+}
+
+/**
+ * Reducer returning the currently autosaving status.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Dispatched action
+ * @return {Object}        Updated state
+ */
+export const autosave = withChangeDetection( function( state = { message: '', autosave: false }, action ) {
+	switch ( action.type ) {
+		case 'RESET_AUTOSAVE':
+			return {
+				...state,
+				autosave: action.autosave
+			}
+		case 'UPDATE_AUTOSAVE_STATUS_MESSAGE':
+			return {
+				...state,
+				message: action.message
+			}
+	}
+
+	return state;
+}, { resetTypes: [ 'UPDATE_AUTOSAVE_STATUS_MESSAGE' ] } );
```

Comment:
@youknowriad when the autosave completes and the status is updated, the state.autosave.isDirty is reset to false